examples/example_weather_agent.py

```diff
@@ -0,0 +1,268 @@
+2. Configuring the worker with weather-related functions
+3. Testing the agent with different cities
+4. Handling API responses and state management
+
+The weather agent can:
+- Fetch current weather for a given city
+- Provide clothing recommendations based on weather
+- Handle multiple cities in sequence
+
+Example:
+    $ export VIRTUALS_API_KEY="your_api_key"
+    $ python examples/example_weather_agent.py
+
+Note:
+    This example requires a valid Virtuals API key to be set in the
+    environment variable VIRTUALS_API_KEY.
+"""
+
+import logging
+import os
+from datetime import datetime
+from typing import Dict, Any, Optional, Tuple
+import requests
+
+from game_sdk.game.agent import Agent
+from game_sdk.game.worker_config import WorkerConfig
+from game_sdk.game.custom_types import Function, Argument
+from game_sdk.game.exceptions import ValidationError, APIError
+
+# Configure logging
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s - %(levelname)s - %(message)s'
+)
+logger = logging.getLogger(__name__)
+
+def get_weather(city: str) -> Dict[str, Any]:
+    """Get weather information for a given city.
+    
+    This function makes an API call to fetch current weather data
+    and provides clothing recommendations based on conditions.
+    
+    Args:
+        city (str): Name of the city to get weather for
+        
+    Returns:
+        Dict[str, Any]: Weather information including:
+            - status: API call status
+            - message: Formatted weather message
+            - data: Detailed weather data
+            
+    Raises:
+        requests.RequestException: If the API call fails
+        
+    Example:
+        >>> result = get_weather("New York")
+        >>> print(result['message'])
+        'Current weather in New York: Sunny, 15°C, 60% humidity'
+    """
+    try:
+        # Simulate API call (replace with actual weather API in production)
+        response = requests.get('https://dylanburkey.com/assets/weather.json')
+        response.raise_for_status()
+        
+        # Process response
+        weather_data = {
+            'New York': {'temp': '15°C', 'condition': 'Sunny', 'humidity': '60%'},
+            'Miami': {'temp': '28°C', 'condition': 'Sunny', 'humidity': '70%'},
+            'Boston': {'temp': '10°C', 'condition': 'Cloudy', 'humidity': '65%'}
+        }.get(city, {'temp': '20°C', 'condition': 'Clear', 'humidity': '50%'})
+        
+        # Determine clothing recommendation
+        temp = int(weather_data['temp'].rstrip('°C'))
+        if temp > 25:
+            clothing = 'Shorts and t-shirt'
+        elif temp > 15:
+            clothing = 'Light jacket'
+        else:
+            clothing = 'Sweater'
+            
+        return {
+            'status': 'success',
+            'message': f"Current weather in {city}: {weather_data['condition']}, "
+                      f"{weather_data['temp']}°F, {weather_data['humidity']} humidity",
```

Comment:
Please fix inconsistency. Weather data uses Celcius but the output message uses Fahrenheit.